preference-v7/src/main/java/com/takisoft/fix/support/v7/preference/PreferenceFragmentCompat.java

```diff
@@ -112,17 +106,25 @@ public void onDisplayPreferenceDialog(Preference preference) {
             } else {
                 super.onDisplayPreferenceDialog(preference);
             }
+        }
+    }
 
-            if (f != null) {
-                if (f instanceof DialogFragment) {
-                    ((DialogFragment) f).setTargetFragment(this, 0);
-                    ((DialogFragment) f).show(this.getFragmentManager(), FRAGMENT_DIALOG_TAG);
-                } else {
-                    this.getFragmentManager()
-                            .beginTransaction()
-                            .add((Fragment) f, FRAGMENT_DIALOG_TAG)
-                            .commit();
-                }
+    protected void displayPreferenceDialog(Fragment fragment, String key) {
+        displayPreferenceDialog(fragment, key, null);
+    }
+    protected void displayPreferenceDialog(Fragment fragment, String key, Bundle bundle) {
+        Bundle b = bundle == null ? new Bundle(1) : bundle;
+        b.putString("key", key);
+        if (fragment != null) {
+            fragment.setArguments(b);
+            if (fragment instanceof DialogFragment) {
+                fragment.setTargetFragment(this, 0);
```

Comment:
If this is going to be merged, can you move `fragment.setTargetFragment(this, 0)` out of the `instanceof DialogFragment` if so that all fragments get a reference to the preference fragment? See https://github.com/Gericop/Android-Support-Preference-V7-Fix/issues/87#issuecomment-339993049 .